Enterprise-CMCS/macfc-quickstart
# Update testing READMEs and scripts

## Purpose

This change updates the top-level README and the Cypress test README to give more information on the types of tests contained in the QuickStart and how to run them. 

In the process, I did some cleanup and reorganization, including:
- standardize script names to use `snake_case` and the `.sh` suffix
- place scripts in their own `scripts` folder in cases where there are multiple scripts in a given folder
- standardize on `yarn` as a runner (vs `npm`) in `package.json` and READMEs
- reorganize the `tests` folder to remove nesting and duplicate `cypress.json` files
  - this included updates to the `tests/cypress` README, the `package.json`, and `cypress.json` itself
- refactor scripts to use the top-level `.github` directory for path references (rather than relative paths, which can break if the folder structure changes)
- remove:
  - commented-out code in testing files
  - scripts that reference dependencies that have since been removed (e.g. `nightwatch`)
 - add:
   -  error handling for scripts that require input but are called without it
   - explanatory comments for scripts
   - Cypress video and screenshot folders to `.gitignore`

Manual Testing: 

- ran all local scripts to make sure they work, including error cases
- ran the `deploy`, `deploySupport`, and `destroy` workflows to make sure that all path references were updated correctly
- ran any updated `package.json` scripts

#### Linked Issues to Close

_Links to issue(s) that are closed by this PR. Be sure to use the phrase "Closes #XXX" for each issue, so they automatically close_

## Approach

_How does this change address the issue?_

## Learning

_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Assorted Notes/Considerations

_List any other information that you think is important... a post-merge activity, someone to notify, etc._

#### Pull Request Creator Checklist


#### Pull Request Reviewer/Assignee Checklist
